dht/dhtcore/Janitor.c

```diff
@@ -264,12 +268,21 @@ static void peersResponseCallback(struct RouterModule_Promise* promise,
         String* addr = Address_toString(from, promise->alloc);
         Log_debug(janitor->logger, "Got peers response from [%s]", addr->bytes);
     }
+
     struct Address_List* addresses =
         ReplySerializer_parse(from, result, janitor->logger, true, promise->alloc);
 
     struct Node_Two* parent = NodeStore_nodeForAddr(janitor->nodeStore, from->ip6.bytes);
     if (!parent) { return; }
 
+    int64_t* moreNodesPtr = Dict_getInt(result, CJDHTConstants_MORE_NODES);
+    uint64_t moreNodes = moreNodesPtr ? *moreNodesPtr : 0;
+    if (moreNodes > 1) {
+        // FIXME(madafoo) malformed getPeers response may cause infinite loop here,
+        // might need threshold control.
+        getPeersMill(janitor, from, moreNodes);
```

Comment:
A way to get rid of that FIXME would be to do a dice-roll which will *almost* always come up true, for example
```c
if (Random_uint8(janitor->rand) != 0xff)
```
which should cause no problems in practice but will break loops if one happens...